src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1225,6 +1261,17 @@ private BoundAssignmentOperator BindAssignment(SyntaxNode node, BoundExpression
                     op2 = conversion;
                 }
 
+                if (refKind != RefKind.None)
+                {
+                    var leftEscape = GetRefEscape(op1, this.LocalScopeDepth);
+                    var rightEscape = GetRefEscape(op2, this.LocalScopeDepth);
```

Comment:
In other places, we followed a different pattern:
```var leftExcape = GetRefEscape(...);
op2 = ValidateRefEscape(op2, leftEscape, ...);```
Then you may not need a new error message.
Or was this intentional so you'd get a specific error message?

Reply:
I intentionally did this to get a better error message. The other error messages say things like, "cannot return this variable because it may expose beyond the lifetime", which makes no sense for assigning one ref local to another.